main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -338,6 +340,38 @@ object RedundancyError {
     })
   }
 
+  /**
+    * An error raised to indicate that a function expression is useless.
+    *
+    * @param tpe the type of the expression.
+    * @param loc the location of the expression.
+    */
+  case class UselessFunctionExpression(tpe: Type, loc: SourceLocation) extends RedundancyError {
+    def summary: String = "Useless function expression."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Useless function expression: It has no side-effect(s) and its result is discarded.
```

Comment:
Under applied function. Missing function argument(s)?

Reply:
Maybe we should go further? Missing argument(s) of type: tpe1, tpe2... 

Lets try that?